src/Nethermind/Nethermind.Synchronization/FastSync/ITreeSync.cs

```diff
@@ -0,0 +1,17 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Synchronization.FastSync;
+
+public interface ITreeSync
+{
+    public event EventHandler<VerifyPostSyncCleanupEventArgs> OnVerifyPostSyncCleanup;
+
+    public class VerifyPostSyncCleanupEventArgs(Hash256 root) : EventArgs
+    {
+        public Hash256 Root => root;
+    }
```

Comment:
I think the names are bad, it is just invoked at the end of sync, `TreeSync` doesn't know what the subscriber will do. Also `On` prefix is for event handlers not events themselves.

```suggestion
    public event EventHandler<SyncCompleatedEventArgs> SyncCompleated;

    public class SyncCompleatedEventArgs(Hash256 root) : EventArgs
    {
        public Hash256 Root => root;
    }
```